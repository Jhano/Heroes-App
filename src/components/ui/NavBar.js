import React, {useContext} from 'react';
import { Navbar, Nav } from 'react-bootstrap';
import { Link, NavLink, useHistory } from 'react-router-dom';
import AuthContext from '../../auth/AuthContext';
import { types } from '../../types/types';




const NavBar = () => {

    //extrayendo la propuedad name y renombrandola
    const {user: {name}, dispatch} = useContext(AuthContext);
    //useHistory, hook para usar history y poder realizar la navegacion cuando es undefine en l componente
    const history = useHistory();

    const handleLogOut = () => {

        dispatch({
            type: types.logout,
        })

        history.replace('/login');
   
    }

    return ( 
        <Navbar bg="dark" expand="md" variant="dark">
            <Navbar.Brand as={Link} to="/">Asociaciones</Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
                <Nav className="mr-auto">
                    <Nav.Link as={NavLink} to="/marvel">Marvel</Nav.Link>
                    <Nav.Link as={NavLink} to="/dc">DC</Nav.Link>
                    <Nav.Link as={NavLink} to="/search">Buscar</Nav.Link>
                </Nav>
                <Nav className="justify-content-end">
                    <Nav.Link  className="text-info">{name}</Nav.Link>
                    <Nav.Link
                        onClick={handleLogOut}
                    >
                        Logout
                    </Nav.Link>
                </Nav>
            </Navbar.Collapse>   
        </Navbar>
     );
}
 
export default NavBar;