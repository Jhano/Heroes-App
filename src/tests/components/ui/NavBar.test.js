import React from 'react';
import '@testing-library/jest-dom'
import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router-dom';

import AuthContext from '../../../auth/AuthContext';
import NavBar from '../../../components/ui/NavBar';
import { types } from '../../../types/types';



describe('Pruebas <NavBar />', () => {

    //se le manda al router y el componente cuando va a usar el useHistory
    //usa el historyMock que tiene se router
    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn()
    }
    
    const contextValue = {
        user: {
            name: 'Alejandro',
            logged: true,
        },
        dispatch: jest.fn()
    }


    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <NavBar/>
                </Router>  
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    })

    test('debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').first().text().trim()).toBe('Alejandro');
    })

    test('debe de llamar el logout y usar history', () => {

       wrapper.find('a').last().simulate('click');

       expect(contextValue.dispatch).toHaveBeenCalledWith({type: types.logout})
       expect(historyMock.replace).toHaveBeenCalledWith('/login');
    })
})