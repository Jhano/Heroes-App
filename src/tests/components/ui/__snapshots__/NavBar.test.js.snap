// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <NavBar /> debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Buscar
      </a>
    </div>
    <div
      className="justify-content-end navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="text-info nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Alejandro
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Logout
      </a>
    </div>
  </div>
</nav>
`;
